src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -467,6 +511,63 @@ internal Region CreateRegion(int startColumn, int startLine, int endColumn = 0,
             return region;
         }
 
+        internal Result CreateResult(ResultVersionOne v1Result)
+        {
+            Result result = null;
+
+            if (v1Result != null)
+            {
+                result = new Result
+                {
+                    AnalysisTarget = CreateFileLocation(v1Result.Locations?[0]?.AnalysisTarget),
+                    BaselineState = Utilities.CreateBaselineState(v1Result.BaselineState),
+                    Id = v1Result.Id,
+                    Level = Utilities.CreateResultLevel(v1Result.Level),
+                    Locations = v1Result.Locations?.Select(CreateLocation).ToList(),
+                    Message = CreateMessage(v1Result.Message),
+                    Properties = v1Result.Properties,
+                    RuleId = v1Result.RuleId,
+                    RuleMessageId = v1Result.FormattedRuleMessage.FormatId,
+                    Stacks = v1Result.Stacks?.Select(CreateStack).ToList(),
+                    SuppressionStates = Utilities.CreateSuppressionStates(v1Result.SuppressionStates),
+                };
+
+                // TODO: transfer formatted rule messages
+
+
+                if (!string.IsNullOrWhiteSpace(v1Result.ToolFingerprintContribution))
+                {
+                    result.PartialFingerprints = new Dictionary<string, string>
+                    {
+                        { "Fingerprint", v1Result.ToolFingerprintContribution }
+                    };
+                }
+
+                // *** Stash un-transferred properties in the property bag *** \\
+
+                if (v1Result.FormattedRuleMessage != null)
+                {
+                    // Only the formattedRuleMessage.formatId property is used above
+                    result.SetProperty($"{FromPropertyBagPrefix}/formattedRuleMessage", v1Result.FormattedRuleMessage);
+                }
+
+                if (!string.IsNullOrWhiteSpace(v1Result.Snippet))
+                {
+                    Region region = result.Locations?[0]?.PhysicalLocation?.Region;
+
+                    if (region != null)
```

Comment:
We have to store the snippet. If the region isn't there, create it. Likewise for its ancestors the physicalLocation object, the location object, and the locations array.